components/automate-ui/src/app/pages/+compliance/shared/reporting/stats.service.ts

```diff
@@ -215,15 +217,20 @@ export class StatsService {
   /* TODO Auth team: Helper functions that are only not private because
      they have unit testing. Should we delete their tests and make
      them private? */
-  getControlStatus(control): string {
+  getControlStatus(control): ControlStatus {
     const statuses = control.results.map(r => r.status);
-    const waived = control.waiver_data && control.waiver_data.skipped_due_to_waiver;
+    const waived = this.checkIfWaived(control.waived_str);
     if (waived) { return 'waived'; }
     if (statuses.every(s => s === 'passed')) { return 'passed'; }
     if (statuses.every(s => s === 'skipped')) { return 'skipped'; }
     return 'failed';
```

Comment:
oooh one thing i just noticed -- some of the other reports we got were about the failure numbers not matching up -- bc the control didnt have any tests, for example. so if a control didnt have tests, it wouldnt have results -- and then it would get "failed" assigned to it, right?  

if any failed = failed
if pass + skip = pass
if all skip = skip
if no results...i guess this could be skip.

Reply:
we can do that in a separate change if ppl wanna hold since that changes the logic a bit more

Reply:
the way its set up now, yeah the default is failed.
sounds like default should be skipped, instead?

Reply:
makes no difference to me - can't merge right now anyways with the license issue.  Let me know what Superteam would like to see and i'll scope it out 👍

Reply:
ok, ya, if you can work out the logic for the failed/pass/skip/waived as described above, that would be awesome. we’ll also make sure the logic in the backend (for the counts) matches this.

Reply:
ok, ya, if you can work out the logic for the failed/pass/skip/waived as described above, that would be awesome. we’ll also make sure the logic in the backend (for the counts) matches this.

Reply:
ah - actually - no tests == passed.

so any failed == failed
all skip == skip
default pass

Reply:
@vjeffrey - Its hard to test since i'm not sure where nodes have missing tests - but if I understand what you're asking for, I think adding a guard clause is a pragmatic approach.